drivers/dma/dma_andes_atcdmac300.h

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2023 Andes Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_DMA_ANDES_ATCDMAC300_H_
+#define ZEPHYR_DRIVERS_DMA_ANDES_ATCDMAC300_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define ATCDMAC100_MAX_CHAN	8
+
+typedef void (*atcdmac300_cfg_func_t)(void);
+
+struct chain_block {
+	uint32_t ctrl;
+	uint32_t transize;
+	uint32_t srcaddrl;
+	uint32_t srcaddrh;
+	uint32_t dstaddrl;
+	uint32_t dstaddrh;
+	uint32_t llpointerl;
+	uint32_t llpointerh;
+	struct chain_block *next_block;
+};
+
+/* data for each DMA channel */
+struct dma_chan_data {
+	void *blkuser_data;
+	dma_callback_t blkcallback;
+	struct chain_block *head_block;
+	struct dma_status status;
+	struct k_spinlock lock;
+};
+
+/* Device run time data */
+struct dma_atcdmac300_data {
+	struct dma_chan_data chan[ATCDMAC100_MAX_CHAN];
+};
+
+/* Device constant configuration parameters */
+struct dma_atcdmac300_cfg {
+	atcdmac300_cfg_func_t irq_config;
+	uint32_t base;
+	uint32_t irq_num;
+};
+
+#define DEV_CFG(dev) ((const struct dma_atcdmac300_cfg * const) (dev)->config)
+#define DEV_DATA(dev) ((struct dma_atcdmac300_data *) (dev)->data)
```

Comment:
We deprecated the usage of these macros.

Reply:
Sorry for that we didn't notice that the macro like DEV_CFG is redundant now.
We have done the modification for it.